packages/client/src/utils/HttpFetcher.ts

```diff
@@ -0,0 +1,27 @@
+import { inject, Lifecycle, scoped } from 'tsyringe'
+import { Debugger } from 'debug'
+import { Context } from './Context'
+import { Config, StrictStreamrClientConfig } from '../Config'
+import { instanceId } from './index'
+import fetch, { Response } from 'node-fetch'
+
+@scoped(Lifecycle.ContainerScoped)
+export class HttpFetcher {
```

Comment:
Replace usages with `Rest#get` and rm this class?

Reply:
Actually looking at `Rest` more closely, it is design to handle session expiration with Core API requests. So not really general enough for this use case.